### PR TITLE
feat: add achievement diary notifier

### LINKED ISSUE #66: Support for Diary completions
image

varbits are our friend
https://static.runelite.net/api/runelite-api/constant-values.html#net.runelite.api.Varbits.DIARY_ARDOUGNE_EASY

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -238,6 +238,26 @@ JSON:
 }
 ```
 
+#### Achievement Diary:
+
+`%AREA%` will be replaced with the geographic area of the achievement diary tasks (e.g., Varrock)
+
+`%DIFFICULTY%` will be replaced with the level of the achievement diary (e.g., Elite)
+
+`%TOTAL%` will be replaced with the total number of achievement diaries completed across locations and difficulties
```
Comment (Felanbird) [inline README.md:247]:
```suggestion
`%TOTAL%` will be replaced with the total number of achievement diaries completed across all locations and difficulties
```

**Comment by Felanbird** [comment]
@pajlada Play with this build doing generic normal things to be sure there aren't any leftover super specific conditions where existing diaries re-fire, very likely all of them were caught during my testing but better to be safe.

**Review by Felanbird — APPROVED** [review]
Functionality works as expected, `diaryMinDifficulty` was tested during debugging.

image
```
{
   "type":"ACHIEVEMENT_DIARY",
   "extra":{
      "area":"Wilderness",
      "difficulty":"MEDIUM",
      "total":32
   },
   "content":"felanbird1 has completed the Medium Wilderness Achievement Diary, for a total of 32 diaries completed",
   "playerName":"felanbird1",
   "embeds":[
      
   ]
}
```
```
{
   "type":"ACHIEVEMENT_DIARY",
   "extra":{
      "area":"Western Provinces",
      "difficulty":"MEDIUM",
      "total":33
   },
   "content":"felanbird1 has completed the Medium Western Provinces Achievement Diary, for a total of 33 diaries completed",
   "playerName":"felanbird1",
   "embeds":[
      
   ]
}
```

**Review by pajlada — APPROVED** [review]
Code looks sane to me, happy to have this merged in since Felanbird has tested it

**Pull request merged**